types/signature_type.go

```diff
@@ -33,5 +35,6 @@ const (
 	EcdsaRecovery   SignatureType = "ecdsa_recovery"
 	Ed25519         SignatureType = "ed25519"
 	Schnorr1        SignatureType = "schnorr_1"
+	SchnorrBip340   SignatureType = "schnorr_bip340"
```

Comment:
Hi @sleepdefic1t , CAN I have more context of why we add it ?
if you wanna support a new type, you can test in at derive endpoint

Reply:
@jingweicb 

> CAN I have more context of why we add it ?

The addition of these types is necessary because the construction and handling of BIP-340 signatures and public keys differ from other signature types.

To reiterate, all of this was previously discussed and approved here: https://github.com/coinbase/mesh-specifications/discussions/113

---

> if you wanna support a new type, you can test in at derive endpoint

Perhaps I'm still misunderstanding.

This PR *only* adds the base signature and curve types as they relate to the __specification__, it's not intended to be a full or partial implementation of BIP-340.

It was discussed and agreed upon that a subsequent PR would handle the actual implementation of BIP-340, which would then utilise these new types.

Could you clarify where you propose adding a test? Without further implementation, testing doesn’t seem possible based solely on the spec update.